node/consensus/src/lib.rs

```diff
@@ -274,14 +290,20 @@ impl<N: Network> Consensus<N> {
         let transactions = {
             // Determine the available capacity.
             let capacity = BatchHeader::<N>::MAX_TRANSMISSIONS_PER_BATCH.saturating_sub(num_unconfirmed);
-            // Acquire the lock on the queue.
-            let mut queue = self.transactions_queue.lock();
-            // Determine the number of transactions to send.
-            let num_transactions = queue.len().min(capacity);
-            // Drain the solutions from the queue.
-            (0..num_transactions)
-                .filter_map(|_| queue.pop_lru().map(|(_, transaction)| transaction))
-                .collect::<Vec<_>>()
+            // Acquire the lock on the deployments queue.
+            let mut deployments_queue = self.deployments_queue.lock();
+            // Acquire the lock on the executions queue.
+            let mut executions_queue = self.executions_queue.lock();
```

Comment:
future consideration: ideally, collections that are always accessed together could be under a single lock (for added performance), though that may require a little bit more type boilerplate

Reply:
Made it anyway so Howard can decide: f93060b9d
Did not implement any further methods for TransactionsQueue to avoid having to add input checking for those methods and to keep it simple.